packages/ngtools/webpack/src/ngcc_processor.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { JsonParseMode, parseJson } from '@angular-devkit/core';
+import * as ts from 'typescript';
+import { time, timeEnd } from './benchmark';
+import { WebpackCompilerHost } from './compiler_host'; // tslint:disable-line:no-circular-imports
+import { workaroundResolve } from './utils';
+
+export class NgccProcessor {
+  private _ngcc: typeof import('@angular/compiler-cli/ngcc') | undefined;
+  private _processedModules = new Set<string>();
+
+  constructor(
+    private readonly propertiesToConsider: string[],
+    private readonly host: WebpackCompilerHost,
+  ) {
+    // this is done for the sole reason that @ngtools/webpack
+    // support versions of Angular that don't have NGCC API
+    try {
+      this._ngcc = require('@angular/compiler-cli/ngcc');
+    } catch {
```

Comment:
Nit: I find this usage pattern a bit odd. So if the require fails, there's a `NgccProcessor` whose `processModule` is a noop. I'd expect there to not be a `NgccProcessor` at all.

Reply:
Updating with my next push.